theano/tensor/opt.py

```diff
@@ -3923,27 +3923,32 @@ def local_elemwise_sub_zeros(node):
 
 @register_canonicalize
 @register_specialize
-@gof.local_optimizer([T.Sum])
-def local_sum_div_dimshuffle(node):
+@gof.local_optimizer([T.Sum, T.elemwise.Prod])
+def local_sum_prod_div_dimshuffle(node):
     '''sum(a / dimshuffle{...}(b), axis=l) -> sum(a, axis={...}) / b,
-    if dimension l of the DimShuffle is 'x'.'''
-    # TODO: extend it to product, and quotient of products
+       if dimension l of the DimShuffle is 'x'
+
+       or
+
+       prod(a / dimshuffle{...}(b), axis=l) -> prod(a, axis={...}) / b,
+      if dimension l of the DimShuffle is 'x'
```

Comment:
I'm not sure that's quite right. Shouldn't that be 
`prod(a / dimshuffle{...}(b), axis=l) -> prod(a, axis={...}) / b ** shape(a)[0]`?
For instance, if a is [[1, 2], [2, 4]], b is [1,2] and l=0 :  
`prod(a / dimshuffle{...}(b), axis=l)` will give vector [2, 2]
`prod(a, axis={...}) / b` will give vector [2, 4]